src/python/py_generate_pipeline.cpp

```diff
@@ -0,0 +1,244 @@
+}
+
+std::string ov_tokenizers_module_path() {
+    // Try a path relative to build artifacts folder first.
+    std::filesystem::path from_relative = ov::genai::tokenizers_relative_to_genai();
+    if (std::filesystem::exists(from_relative)) {
+        return from_relative.string();
+    }
+    return py::str(py::module_::import("openvino_tokenizers").attr("_ext_path"));
+}
+
+class EmptyStreamer: public StreamerBase {
+    // It's impossible to create an instance of pure virtual class. Define EmptyStreamer instead.
+    bool put(int64_t token) override {
+        PYBIND11_OVERRIDE_PURE(
+            bool,  // Return type
+            StreamerBase,  // Parent class
+            put,  // Name of function in C++ (must match Python name)
+            token  // Argument(s)
+        );
+    }
+    void end() override {
+        PYBIND11_OVERRIDE_PURE(void, StreamerBase, end);
+    }
+};
+
+ov::InferRequest& get_request_from_pyobj(py::object obj) {
+    py::str obj_type = py::str(obj.get_type());
+    // todo: InferRequest is not accessible from the outside.
+    // obj_type is openvino._pyopenvino.InferRequest,
+    // which is a pybind binding to InferRequestWrapper (InferRequest is in a m_request field of the latest)
+    // and the definition of InferRequestWrapper is not accessible from the outside.
+
+    if (py::isinstance<ov::InferRequest>(obj)) {
+        // Directly return the casted object without copying
+        return obj.cast<ov::InferRequest&>();
+    } else {
+        throw std::invalid_argument("Provided object is not castable to ov::InferRequest");
+    }
+}
+
+} // namespace
+
+
+PYBIND11_MODULE(py_generate_pipeline, m) {
+    m.doc() = "Pybind11 binding for LLM Pipeline";
+
+    py::class_<LLMPipeline>(m, "LLMPipeline")
+        .def(py::init([](const std::string& model_path, 
+                            const std::string& device) {
+            ov::genai::ScopedVar env_manager(ov_tokenizers_module_path());
+            return std::make_unique<LLMPipeline>(model_path, device);}),
+        py::arg("model_path"), "path to the model path", 
+        py::arg("device") = "CPU", "device on which inference will be done",
+        R"(
+            LLMPipeline class constructor.
+            model_path (str): Path to the model file.
+            device (str): Device to run the model on (e.g., CPU, GPU). Default is 'CPU'.
+        )")
+
+        .def(py::init<const std::string, const Tokenizer&, const std::string>(), 
+        py::arg("model_path"),
+        py::arg("tokenizer"),
+        py::arg("device") = "CPU",
+        R"(
+            LLMPipeline class constructor for manualy created openvino_genai.Tokenizer.
+            model_path (str): Path to the model file.
+            tokenizer (openvino_genai.Tokenizer): tokenizer object.
+            device (str): Device to run the model on (e.g., CPU, GPU). Default is 'CPU'.
+        )")
+
+        .def(py::init([](py::object infer_request, 
```

Comment:
I tried to create an infer request from outside with core.compile_model().create_infer_request() and put it here and got an error:
```
TypeError: __init__(): incompatible constructor arguments. The following argument types are supported:
    1. openvino_genai.py_generate_pipeline.LLMPipeline(model_path: str, device: str = 'CPU')
    2. openvino_genai.py_generate_pipeline.LLMPipeline(model_path: str, tokenizer: ov::genai::Tokenizer, device: str = 'CPU')
    3. openvino_genai.py_generate_pipeline.LLMPipeline(infer_request: object, tokenizer: ov::genai::Tokenizer, config: Optional[ov::genai::GenerationConfig])
```

Is it right behavior ? How could I create the infer_requst to put it here ?

Reply:
The exact python line could help. The error message suggests, it's possible to do, but the args were incorrect:
`3. openvino_genai.py_generate_pipeline.LLMPipeline(infer_request: object, tokenizer: ov::genai::Tokenizer, config: Optional[ov::genai::GenerationConfig])`